src/ApiService/ApiService/OneFuzzTypes/Events.cs

```diff
@@ -52,12 +50,42 @@ this switch
             {
                 EventNodeHeartbeat _ => EventType.NodeHeartbeat,
                 EventInstanceConfigUpdated _ => EventType.InstanceConfigUpdated,
+                EventTaskHeartbeat _ =>  EventType.TaskHeartbeat,
                 _ => throw new NotImplementedException(),
             };
 
     }
 };
 
+public class BaseEventConverter : JsonConverter<BaseEvent>
+{
+    public override BaseEvent? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        return null;
+    }
+
+    public override void Write(Utf8JsonWriter writer, BaseEvent value, JsonSerializerOptions options)
+    {
+        var eventType = value.GetType();
+        JsonSerializer.Serialize(writer, value, eventType, options);
+    }
+}
+
+public class EventTypeProvider : ITypeProvider
+{
+    public Type GetTypeInfo(object input)
+    {
+        return (input as EventType?) switch
+        {
+            EventType.NodeHeartbeat => typeof(EventNodeHeartbeat),
```

Comment:
can we have one centralized function for this ? so we just update this switch in one place. 

Otherwise we going to forget to update one of those for sure

Reply:
moved the logic to BaseEvent